Fix version of `web3` to resolve BC breaks

spirit-q2 opened the pull request:
we have to lock `web3` at `v5.20.0` since after that version they switched to `websockets` of `v9.1` which brings BC break (comparing to `v8.1`) to our code by deprecating some classes that we use. I.e. `websockets.protocol` became `websockets.legacy.protocol`.

Related to #272 


┆Issue is synchronized with this [Asana task](https://app.asana.com/0/1200560623420491/1201298108242295) by [Unito](https://www.unito.io)

spirit-q2 committed:
Add `web3` to required packages

CLAassistant commented:
[![CLA assistant check](https://cla-assistant.io/pull/badge/signed)](https://cla-assistant.io/humanprotocol/hmt-escrow?pullRequest=281) <br/>All committers have signed the CLA.

spirit-q2 committed:
Update `Pipfile.lock` with `pipenv lock` command

spirit-q2 committed:
Fix `mypy` error

Argument "shared_mac_data" to "decrypt" has incompatible type "Union[str, bytes]"; expected "bytes"
Argument "shared_mac_data" to "encrypt" has incompatible type "Union[str, bytes]"; expected "bytes"

spirit-q2 committed:
Display `black` errors, if any.

rustyro commented on hmt_escrow/storage.py:
```diff
@@ -196,7 +201,12 @@ def _encrypt(public_key: bytes, msg: str) -> bytes:
+        SHARED_MAC_DATA
+        if isinstance(SHARED_MAC_DATA, bytes)
+        else SHARED_MAC_DATA.encode("ascii")
+    )
```
@spirit-q2 can this be done at the top of the file, so it can be reused here and in the reference above w/o the repetition?

spirit-q2 replied:
found nice solution :slightly_smiling_face:

spirit-q2 replied:
seems to be working :tada:

e271828- commented:
Why don't we simply do a compatible import, i.e.

```
try:
    from x.websockets import protocol as websocketsprotocol
except:
    from x.legacy.websockets import protocol as websocketsprotocol
```

?

spirit-q2 commented:
@e271828-, there might be other incompatibilities as well. Although, there is a list of BC breaks: https://websockets.readthedocs.io/en/latest/project/changelog.html#id11 :thinking: 
I can do some investigation on the usage of code that was moved/deprecated if you think it's worth spending time on it.

spirit-q2 committed:
Optimize fix for `mypy`

e271828- commented:
> @e271828-, there might be other incompatibilities as well. Although, there is a list of BC breaks: https://websockets.readthedocs.io/en/latest/project/changelog.html#id11 🤔 I can do some investigation on the usage of code that was moved/deprecated if you think it's worth spending time on it.

Seems like we should at minimum switch to use the new refs so we can upgrade. Pinning to an arbitrary stale version with no ability to move forward short of code changes will make any security-related upgrades challenging.

spirit-q2 commented:
Makes sense 👍 will do. I like your approach for upgrading the versions :)

spirit-q2 commented:
@e271828- we are done here, since the `hmt-escrow` don't use the `websockets` package directly. BC breaks should be handled on our side. Please review/publish if all fine in the PR
